### PR TITLE
chore: Use navigation options to modify header

### PR DESCRIPTION
**Pull request by bruce-glazier**
## Changes

Respect some header navigation options in our custom navigation header:
- headerTitle: Override title with either a title string or components
- headerLeft: Inject buttons/etc into the header in the left position
- headerRight: Inject buttons/etc into the header in the right position
- headerShown: Hide the header as needed

## Screenshots

https://github.com/lifeomic/react-native-sdk/assets/76954025/cbfe1c0b-1fbc-47b0-85f9-34983c6476a0

### CODE REVIEW

File: src/components/AppNavHeader.tsx [hunk 1]
```diff
@@ -22,7 +23,15 @@ export function AppNavHeader({
 }: NativeStackHeaderProps) {
   const { styles } = useStyles(defaultStyles);
   const { ChevronLeft } = useIcons();
-  const title = options.title || route.name;
+  const { headerTitle } = options;
+  const { headerString, headerFunction } =
+    typeof headerTitle === 'string'
```
Comment (sternetj) [inline src/components/AppNavHeader.tsx:28]: I think the type of `headerTitle` is `string | function` so you could probably reverse this and be able to remove the second ternary.

**Review by sternetj — APPROVED** [review]

File: src/components/AppNavHeader.tsx (hunk 1 above)
Comment (bruce-glazier) [reply src/components/AppNavHeader.tsx:28]: 👍 Good point

File: src/components/AppNavHeader.tsx [hunk 2]
```diff
@@ -65,13 +78,35 @@ export function AppNavHeader({
           style={styles.backAction}
         />
       ) : null}
+      {options.headerLeft && (
```
[the remark this answers is not in the document]
Comment (bruce-glazier) [reply src/components/AppNavHeader.tsx:81]: I wasn't sure if there was ever a scenario where someone could want both but I suppose the could always pass a back button into the header left function.

[the remark this answers is not in the document]
Comment (sternetj) [reply src/components/AppNavHeader.tsx:81]: Yeah that's kind of what I was thinking. If they want to override the back button then they need to provide their own if they want one.

File: src/components/AppNavHeader.tsx [hunk 3]
```diff
@@ -65,13 +78,35 @@ export function AppNavHeader({
...
+            <Title text={title} style={titleStyles} />
+          )
+        }
         style={[styles.contentView, back && styles.contentViewWithBackButton]}
       />
-      {options.headerRight?.({
-        canGoBack: !!back,
-      })}
+      {options.headerRight && (
+        <View>
```
Comment (markdlv) [inline src/components/AppNavHeader.tsx:103]: Just curious, why was this extra View needed?  Another approach could be, if/when it is needed, it's provided inside the `headerRight` func provided.  But I'm guessing we're betting this View will be needed 99% of the time?

File: src/components/AppNavHeader.tsx [hunk 4]
```diff
@@ -65,13 +78,35 @@ export function AppNavHeader({
           style={styles.backAction}
         />
       ) : null}
+      {options.headerLeft && (
+        <View>
+          {options?.headerLeft?.({
+            tintColor: options.headerTintColor,
+            canGoBack: !!back,
+          })}
+        </View>
```
Comment (markdlv) [inline src/components/AppNavHeader.tsx:87]: Same thing goes here as below - why wrap `headerLeft` in this `View`?

File: src/components/AppNavHeader.tsx (hunk 3 above)
Comment (bruce-glazier) [reply src/components/AppNavHeader.tsx:103]: I had intended to do something else with this view that I never did implement, I can probably drop it.

**Review by sternetj — APPROVED** [review]

**Pull request merged**